test/general/backend_inclusion.pass.cpp

```diff
@@ -0,0 +1,54 @@
+// -*- C++ -*-
+//===-- backend_inclusion.pass.cpp ----------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+
+#include "support/utils.h"
+
+#if !_ONEDPL_PAR_BACKEND_SERIAL
+#   ifdef _ONEDPL_PARALLEL_BACKEND_SERIAL_H
+#        error The parallel serial backend is used while it should not (_ONEDPL_PAR_BACKEND_SERIAL==0)
+#   endif
+#endif
+
+#if !_ONEDPL_PAR_BACKEND_OPENMP
+#   ifdef _ONEDPL_PARALLEL_BACKEND_OMP_H
+#       error The parallel OpenMP backend is used while it should not (_ONEDPL_PAR_BACKEND_OPENMP==0)
+#   endif
+#endif

```

Comment:
I think it's better to use public macros (`ONEDPL_USE_*`) if available, instead of internal `_ONEDPL_*`macros.

Reply:
Are you sure? I believe the idea was if we somehow defined that one or the other backend should be taken by our internal logic, we check that we don't include something wrong.

For example we allow the user to do not define any public macro. In that case TBB backend is taken by default. IICU, with what you suggested the test will not pass.

On the other hand with you comment I start thinking about the "correctness" of this test. I mean the following.

Let's take the `_ONEDPL_PAR_BACKEND_OPENMP` as the reference. If this macro is not defined or zero we definitely should not observe `_ONEDPL_PARALLEL_BACKEND_OMP_H`. On the other hand, if  `_ONEDPL_PAR_BACKEND_OPENMP` is 1 currently there are circumstances when we also should not observe `_ONEDPL_PARALLEL_BACKEND_OMP_H`.

Should we check that in this test? or should it be separate test?

Reply:
The checks are made with using public macros where possible.